src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -260,13 +300,15 @@ private FSMState nextState(TeleopInput input) {
 				if (finishedPurePursuitPath) {
 					finishedPurePursuitPath = false;
 					ppController = new PurePursuit(pointsToHub);
+					timer.reset();
```

Comment:
Are these resets necessary?

Reply:
We added a time-based safety check to the paths (basically if the state hasn't moved on to the next one in a certain amount of time, it goes to the next state). We did this just incase the robot hits a wall and isn't able to complete the path. This is why we reset the timer during the state transition (yes, we should rename the timer).

Reply:
But the timer gets reset in update() after 1 second. Might want to double check the logic. BTW this kind of safety timer is often called a "watchdog" timer.